js/config.js

```diff
@@ -214,128 +217,98 @@ function saveUserConfig(fieldName, newValue, oldConfig) {
+                resolve(updateUserConfig(data[0]));
             })
             .catch((err) => {
                 reject(err);
             });
+
     });
+
 }
 
 /**
- * Method to overwrite user config on mac installer
+ * Manipulates user config on macOS
  * @param {String} globalConfigPath - The global config path from installer
  * @returns {Promise}
  */
-function updateUserConfigMac(globalConfigPath) {
+function updateUserConfigMac() {
     return new Promise((resolve, reject) => {
         const userConfigFile = path.join(dirs.userConfig(), configFileName);
 
-        // if user config file does't exists just copy global config file
+        // if user config file does't exist, just use the global config settings
+        // i.e. until an user makes changes manually using the menu items
         if (!fs.existsSync(userConfigFile)) {
-            resolve(copyConfigMac(globalConfigPath));
+            log.send(logLevels.WARN, 'config: Could not find the user config file!');
+            reject();
             return;
         }
 
-        Promise.all([readGlobalConfig(), readUserConfig(userConfigFile)])
-            .then((data) => {
-                resolve(updateUserConfig(data[0], data[1]));
-            })
-            .catch((err) => {
-                reject(err);
-            });
-    });
-}
-
-/**
- * Method to copy global config file to user config directory for Windows
- * @returns {Promise}
- */
-function copyConfigWin() {
-    return new Promise((resolve, reject) => {
-        const globalConfigFileName = path.join('config', configFileName);
-        const execPath = path.dirname(app.getPath('exe'));
-        const globalConfigPath = path.join(execPath, '', globalConfigFileName);
-        const userConfigPath = app.getPath('userData');
-
-        childProcess.exec(`echo D|xcopy /y /e /s /c "${globalConfigPath}" "${userConfigPath}"`, { timeout: 60000 }, (err) => {
-            if (err) {
-                reject(err);
-                return;
-            }
-            resolve();
-        });
-    });
-}
-
-/**
- * Method which copies global config file to user config directory for mac
- * @param {String} globalConfigPath - The global config path from installer
- * @returns {Promise}
- */
-function copyConfigMac(globalConfigPath) {
-    return new Promise((resolve, reject) => {
-        let userConfigPath = dirs.userConfig() + '/';
-        let userName = process.env.USER;
-
-        childProcess.exec(`rsync -r "${globalConfigPath}" "${userConfigPath}" && chown -R "${userName}" "${userConfigPath}"`, { timeout: 60000 }, (err) => {
-            if (err) {
-                reject(err);
-                return;
-            }
-            resolve();
+        // In case the file exists, we remove it so that all the
+        // values are fetched from the global config
+        // https://perzoinc.atlassian.net/browse/ELECTRON-126
+        Promise.all([readUserConfig(userConfigFile)])
```

Comment:
same comment as above.

Reply:
Not really, I can remove it. Since we had multiple functions being called before and for future purposes, just left it as is.

I'll remove it for now.